src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int BIT_SHIFT_VALUE = 1;
+    private static final int ONE_ELEMENT_SIZE = 1;
+    private static final int EMPTY_LIST_VALUE = 0;
+    private static final int FIRST_ELEMENT_INDEX = 0;
+    private static final int NO_SUCH_ELEMENT_INDEX = -1;
+    private static final String NO_SUCH_ELEMENT_MESSAGE = "There is no such element in list.";
+    private static final String INDEX_OUT_OF_BOUNDS_MESSAGE = "This index does not exist in list.";
+
+    private Object[] data;
```

Comment:
you can create an array of generics

Reply:
I think it is not necessary because when will runtime be it will resolves to Object, so we can declare Object array instead of generics array

![image](https://github.com/mate-academy/jv-array-list/assets/137708693/2f311e20-ca7d-4ffa-9849-f6b85b5cfc1a)